src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvoker.cs

```diff
@@ -672,12 +690,15 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
 
                         if (exceptionContext != null)
                         {
-                            if (exceptionContext.Result != null && !exceptionContext.ExceptionHandled)
+                            if (exceptionContext.Result != null ||
+                                exceptionContext.Exception == null ||
+                                exceptionContext.ExceptionHandled)
```

Comment:
Follow up: I was incorrect and the ASP.NET Core MVC 1.0.0 behaviour when an `IExceptionFilter` sets `Result` but neither nulls out `Exception` nor sets `ExceptionHandled` is not the same as MVC 5.x: ASP.NET Core 1.0.0 and 1.1.0 both execute the `Result` in this case and neither short-circuits.